File: backend/src/main/java/com/cybercrimeanalytics/service/ComplaintService.java
package com.cybercrimeanalytics.service;

import com.cybercrimeanalytics.dto.request.ComplaintRequest;
import com.cybercrimeanalytics.dto.request.ComplaintSearchRequest;
import com.cybercrimeanalytics.dto.response.ComplaintResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.UUID;

public interface ComplaintService {
    ComplaintResponse create(ComplaintRequest request, String username);
    ComplaintResponse update(UUID id, ComplaintRequest request);
    void delete(UUID id);
    ComplaintResponse get(UUID id);
    Page<ComplaintResponse> search(ComplaintSearchRequest request, Pageable pageable);
}

File: frontend/src/services/complaintService.js
import { api } from '../api/client';

export const complaintService = {
  list: (params = {}) => api.get('/complaints', { params }).then((r) => r.data),
  get: (id) => api.get(`/complaints/${id}`).then((r) => r.data),
  create: (payload) => api.post('/complaints', payload).then((r) => r.data),
  update: (id, payload) => api.put(`/complaints/${id}`, payload).then((r) => r.data),
  remove: (id) => api.delete(`/complaints/${id}`)
};
